src/Sema.zig

```diff
@@ -38426,6 +38420,13 @@ fn resolveDeclaredEnumInner(
             return sema.failWithOwnedErrorMsg(block, msg);
         }
     }
+    if (small.nonexhaustive and int_tag_ty.toIntern() != .comptime_int_type) {
+        if (fields_len > 1 and std.math.log2_int(u64, fields_len) == int_tag_ty.bitSize(zcu) or
+            fields_len >= 1 and int_tag_ty.bitSize(zcu) == 0)
+        {
```

Comment:
Why is this two checks rather than just:
```suggestion
        if (fields_len >= 1 and std.math.log2_int(u64, fields_len) == int_tag_ty.bitSize(zcu)) {
```

Reply:
Not sure what I was thinking when I originally wrote this, you're absolutely right.